src/dodal/devices/motors.py

```diff
@@ -2,60 +2,173 @@
+        z_infix: str = "Z",
     ):
         with self.add_children_as_readables():
-            self.x = Motor(prefix + infix[0])
-            self.y = Motor(prefix + infix[1])
-            self.z = Motor(prefix + infix[2])
-        super().__init__(name=name)
+            self.z = Motor(prefix + z_infix)
+        super().__init__(prefix, name, x_infix, y_infix)
 
 
-class SixAxisGonio(XYZPositioner):
+class XYZThetaStage(XYZStage):
+    """
+
+    ophyd_async x y z theta motor stage, combining 3  mutually perpendicular Motors and
+    a rotational stage in theta.
+
+    Parameters
+    ----------
+    prefix:
+        Common part of the EPICS PV for all motors, including ":".
+    name:
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
+
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        name: str = "",
+        x_infix: str = "X",
+        y_infix: str = "Y",
+        z_infix: str = "Z",
+        theta_infix: str = "Z",
+    ) -> None:
+        with self.add_children_as_readables():
+            self.theta = Motor(prefix + theta_infix)
+        super().__init__(prefix, name, x_infix, y_infix, z_infix)
+
+
+class SixAxisGonio(XYZStage):
+    """
+
+    ophyd_async x y z kappa phi omega motor stage, combining 3  mutually perpendicular
+    Motors and 3 mutually perpendicular rotational stages.
+
+    Parameters
+    ----------
+    prefix:
+        Common part of the EPICS PV for all motors, including ":".
+    name:
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
+
+    """
+
     def __init__(
         self,
         prefix: str,
         name: str = "",
-        infix: tuple[str, str, str, str, str, str] = (
-            "X",
-            "Y",
-            "Z",
-            "KAPPA",
-            "PHI",
-            "OMEGA",
-        ),
+        x_infix: str = "X",
+        y_infix: str = "Y",
+        z_infix: str = "Z",
+        kappa_infix: str = "KAPPA",
+        phi_infix: str = "PHI",
+        omega_infix: str = "OMEGA",
     ):
         with self.add_children_as_readables():
-            self.kappa = Motor(prefix + infix[3])
-            self.phi = Motor(prefix + infix[4])
-            self.omega = Motor(prefix + infix[5])
-        super().__init__(name=name, prefix=prefix, infix=infix[0:3])
+            self.kappa = Motor(prefix + kappa_infix)
+            self.phi = Motor(prefix + phi_infix)
+            self.omega = Motor(prefix + omega_infix)
+        super().__init__(prefix, name, x_infix, y_infix, z_infix)
+
+
+class YZStage(StandardReadable):
+    """Physical motion for detector travel
+
+    ophyd_async y z motor stage, combining 2 Motors: a horizontal stage parallel to the
+    beam and a vertical stage perpendicular to it.
```

Comment:
Nit: It's not immediately clear which of the description refers to y and which to z. Again, I would also say something like "usually"